Autopilot/PathManager/Inc/pathManagerStateManager.hpp

```diff
@@ -0,0 +1,26 @@
+/**
+ *  Path Manager State Machine Header
+ */
+#pragma once
+
+#ifndef PATHMANAGERSTATEMANAGER_HPP
+#define PATHMANAGERSTATEMANAGER_HPP
+
+#include "pathManager.hpp"
+
+class pathManager; //WHY HAVE THIS HERE?!?!? !!QUESTION!!
+
+class pathManagerState
+{
+    public:
+        virtual void enter(pathManager* pathMgr) = 0;
+        virtual void execute(pathManager* pathMgr) = 0;
+        virtual void exit(attitudeManager* pathMgr) = 0;
+
+        bool operator==(const pathManagerState& rhs) const {return (this == &rhs);} //WHAT IS THIS?? !!QUESTION!!
```

Comment:
This is overloading the "==" operator to check if the memory address of our current state is correct. Since we are looking at class objects, the regular "==" would not work since it checks values (not memory address locations). By overriding, we create this custom functionality that is great for testing and stuff!